src/index.ts

```diff
@@ -140,27 +29,61 @@ export default function reactDocgenTypescript(config: Options = {}): Plugin {
       )
     )
     .reduce((carry, files) => carry.concat(files), []);
-  const tsProgram = ts.createProgram(files, compilerOptions);
+
+  const result = {
+    ts,
+    files,
+    compilerOptions,
+    docGenParser,
+    filter,
+    generateOptions,
+    generateDocgenCodeBlock,
+  };
+
+  return result;
+};
+
+export default function reactDocgenTypescript(config: Options = {}): Plugin {
+  const utilsPromise = getUtils(config);
 
   return {
     name: "vite:react-docgen-typescript",
     async transform(src, id) {
-      if (!filter(id)) return;
-      const componentDocs = docGenParser.parseWithProgramProvider(
-        id,
-        () => tsProgram
-      );
-
-      if (!componentDocs.length) {
-        return null;
+      try {
+        const {
+          ts,
+          filter,
+          files,
+          docGenParser,
+          compilerOptions,
+          generateOptions,
+          generateDocgenCodeBlock,
+        } = await utilsPromise;
+
+        const tsProgram = ts.createProgram(files, compilerOptions);
+
+        if (!filter(id)) {
+          return;
+        }
+
+        const componentDocs = docGenParser.parseWithProgramProvider(
+          id,
+          () => tsProgram
+        );
+
+        if (!componentDocs.length) {
+          return null;
+        }
+
+        return generateDocgenCodeBlock({
+          filename: id,
+          source: src,
+          componentDocs,
+          ...generateOptions,
+        });
+      } catch (e) {
+        return src;
```

Comment:
Should we print a warning if this fails?  Might be good to alert the user that it failed.

Reply:
I'm not sure, if so, it should only log once total, not once for every file.

Does vite have some api for reporting problems? If so that would be a lot nicer than rolling our own "dump a bunch of crap" into the terminal.